### DISCUSSION TITLE
Archiving HackerNews links

### DISCUSSION THREAD

**Discussion by diego898** [ArchiveBox/ArchiveBox#913] [Q&A]
Hello,

I have a long list of HackerNews links that I'd like to archive. For each link I'd like to archive the HN page (comments/discussions), and the thing the HN page links to. 

As I understand it, Archivebox cannot do this itself and I must use another tool as well. Has anyone else done something similar?

**Comment by pirate** [comment]
You can do with this with archivebox itself, e.g. `archivebox add --depth=1 < list_of_hn_links.txt`. It will archive the discussion page + all outlinks, which includes the article itself (along with anything linked to in the comments).

It may archive more than what you need, but you'll get those as a minimum.

**Reply by diego898** [reply]
Thanks for the reply! This seems to generate way to many links and 1 article has now corrupted my archive.

As an example, just this one `https://news.ycombinator.com/item?id=28607995` generated over 5000 links.

Clicking on them in my archive, there are now separate entries for each comment, etc.

**Reply by pirate** [reply]
Ah you may have to filter those with the BLACKLIST_URLS or WHITELIST_URLS config option.

**Reply by diego898** [reply]
Hm I'm not actually sure that would work as each article and comment on the article have their own URL of the same format, but different ids. IE: the article is: `https://news.ycombinator.com/item?id=28607995 ` and its first comment is `https://news.ycombinator.com/item?id=28609487` and the child is `https://news.ycombinator.com/item?id=28613913` etc. 

So I wouldn't know what you populate those with.

Is it possible to restrict the depth by DOM element? Like, only follow the `main-link` element? 

(I don't actually know the name of that link, or if it that changes, etc - just using an example)

**Reply by pirate** [reply]
Ah that is a pain, I don't have an easy solution then. Unless there is some way to distinguish via regex you may have to settle for `archivebox remove ...` afterwards or using a more powerful crawler like `scrapy` and piping the outputted URLs into archivebox.
